solr/views.py

```diff
@@ -178,6 +185,93 @@ def _get_host(self, url):
         else:
             self._host = parts[0]
         return self._host
+    
+    def _extract_docs_values(self, input):
+        out = []
+        i = 0
+        while input.find('docs(', i) > -1:
+            i = input.index('docs(', i) + 5
+            j = i
+            while input[j] != ')' and j < len(input):
+                j += 1
+            out.append(input[i:j])
+            i = j + 1
+        return out
+    
+    def check_for_embedded_bigquery(self, params, request):
+        """Checks for the presence of docs() query any where inside 
+        the query parameters; if present - we'll verify/update
+        the query with data.
+        """
+        streams = set()
+        for k,v in params.items():
+            if 'q' in k: # well, i was laying - we'll only check params that *could* be a query
+                if isinstance(v, basestring):
+                    if 'docs(' in v:
+                        streams.update(self._extract_docs_values(v))
+                else:
+                    for x in v:
+                        if 'docs(' in x:
+                            streams.update(self._extract_docs_values(x))
+                
+        if len(streams) == 0:
+            return
+        
+        # must verify the input is there
+        for sn in streams:
+            if sn in params: # it can be in the parameters, which is OK...
```

Comment:
When would the docs be in the params?

Reply:
if one decided to save params not in the bigquery (stream) but inside url params; that's the only case when docs() can be used against /search/query endpoint